distributed/scheduler.py

```diff
@@ -422,6 +426,17 @@ def start(self, addr_or_port=8786, start_queues=True):
             for k, v in self.services.items():
                 logger.info("%11s at: %25s", k, '%s:%d' % (self.ip, v.port))
 
+        if self.scheduler_file:
+            with ignoring(ImportError):
```

Comment:
I think it would be nice to refactor scheduler file handling into separate functions (see also client.py and dask_worker.py).

Also why the `ImportError`?